uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/controller/UniProtKBSearchControllerIT.java

```diff
@@ -1426,6 +1427,68 @@ void searchDefaultSearchWithIsoformLowercase() throws Exception {
                         jsonPath("$.results.*.primaryAccession", containsInAnyOrder("P21802-2")));
     }
 
+    @Test
+    void searchDefaultSearchRheaUpperCase() throws Exception {
+        // given
+        UniProtKBEntry entry = UniProtEntryMocker.create(UniProtEntryMocker.Type.SP_CANONICAL);
+        getStoreManager().save(DataStoreManager.StoreType.UNIPROT, entry);
+
+        entry = UniProtEntryMocker.create(UniProtEntryMocker.Type.TR);
+        // Add rhea id that can be found as default search
+        UniProtKBEntryBuilder eb = UniProtKBEntryBuilder.from(entry);
+        ProteinDescriptionBuilder pd =
+                ProteinDescriptionBuilder.from(entry.getProteinDescription());
+        pd.allergenName(new NameBuilder().value("Injected RHEA:10596").build());
+        eb.proteinDescription(pd.build());
+        getStoreManager().save(DataStoreManager.StoreType.UNIPROT, eb.build());
+
+        // when
+        ResultActions response =
+                getMockMvc()
+                        .perform(
+                                get(SEARCH_RESOURCE)
+                                        .param("query", "RHEA:10596")
+                                        .header(ACCEPT, APPLICATION_JSON_VALUE));
+
+        // then
+        response.andDo(print())
+                .andExpect(status().is(HttpStatus.OK.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(jsonPath("$.results.size()", is(1)))
+                .andExpect(jsonPath("$.results[0].primaryAccession", is("P21802")));
+    }
+
+    @Test
+    void searchDefaultSearchWithUnderscoreIds() throws Exception {
+        // given
+        UniProtKBEntry entry = UniProtEntryMocker.create(UniProtEntryMocker.Type.SP_CANONICAL);
+        getStoreManager().save(DataStoreManager.StoreType.UNIPROT, entry);
+
+        entry = UniProtEntryMocker.create(UniProtEntryMocker.Type.TR);
+        // Add rhea id that can be found as default search
+        UniProtKBEntryBuilder eb = UniProtKBEntryBuilder.from(entry);
+        ProteinDescriptionBuilder pd =
+                ProteinDescriptionBuilder.from(entry.getProteinDescription());
+        pd.allergenName(new NameBuilder().value("Injected VAR test 004127").build());
+        eb.proteinDescription(pd.build());
+        getStoreManager().save(DataStoreManager.StoreType.UNIPROT, eb.build());
+
```

Comment:
This test case is fine but i wish we had one more entry where numerical part of var was in some other section like we have VSP_035901 in https://rest.uniprot.org/uniprotkb/A6QNS3.txt . But i know the fix will work